<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Knot_Window</class>
 <widget class="QMainWindow" name="Knot_Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Knotter</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/img/logo.svg</normaloff>:/img/logo.svg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="KnotView" name="canvas"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>640</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="actionSave_As"/>
    <addaction name="action_Export"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menuNo_des">
    <property name="title">
     <string>No&amp;des</string>
    </property>
    <addaction name="action_Link"/>
    <addaction name="action_Erase"/>
    <addaction name="action_Merge"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="action_Horizontal_Flip"/>
    <addaction name="action_Vertical_Flip"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="action_Undo"/>
    <addaction name="action_Redo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="action_Copy"/>
    <addaction name="action_Paste"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="action_Configure_Knotter"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <widget class="QMenu" name="menu_Zoom">
     <property name="title">
      <string>&amp;Zoom</string>
     </property>
     <addaction name="actionZoom_In"/>
     <addaction name="actionZoom_Out"/>
     <addaction name="action_Reset_Zoom"/>
    </widget>
    <widget class="QMenu" name="menu_Toolbars">
     <property name="title">
      <string>&amp;Toolbars</string>
     </property>
     <addaction name="action_Main"/>
     <addaction name="action_Knot_Editing"/>
    </widget>
    <addaction name="menu_Toolbars"/>
    <addaction name="separator"/>
    <addaction name="menu_Zoom"/>
    <addaction name="action_Reset_View"/>
    <addaction name="separator"/>
    <addaction name="actionShow_Graph"/>
    <addaction name="actionShow_Knot_line"/>
    <addaction name="separator"/>
    <addaction name="actionEnable_Grid"/>
    <addaction name="actionConfigure_G_rid"/>
    <addaction name="action_Move_Grid"/>
   </widget>
   <widget class="QMenu" name="menu_Knot">
    <property name="title">
     <string>&amp;Knot</string>
    </property>
    <addaction name="action_Style"/>
    <addaction name="separator"/>
    <addaction name="actionInsert_Edges"/>
    <addaction name="actionEdge_List"/>
    <addaction name="separator"/>
    <addaction name="actionRefresh_Path"/>
   </widget>
   <widget class="QMenu" name="menu_Debug">
    <property name="title">
     <string>&amp;Debug</string>
    </property>
    <addaction name="actionForce_Crash"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
   <addaction name="menu_View"/>
   <addaction name="menuNo_des"/>
   <addaction name="menu_Knot"/>
   <addaction name="menu_Debug"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="EditBar">
   <property name="windowTitle">
    <string>EditBar</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllToolBarAreas</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionInsert_Edges"/>
   <addaction name="actionEdge_List"/>
   <addaction name="separator"/>
   <addaction name="action_Link"/>
   <addaction name="action_Erase"/>
   <addaction name="action_Merge"/>
   <addaction name="separator"/>
   <addaction name="actionEnable_Grid"/>
   <addaction name="separator"/>
   <addaction name="action_Horizontal_Flip"/>
   <addaction name="action_Vertical_Flip"/>
  </widget>
  <widget class="QToolBar" name="MainToolBar">
   <property name="windowTitle">
    <string>MainToolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="actionSave_As"/>
   <addaction name="action_Export"/>
   <addaction name="separator"/>
   <addaction name="action_Undo"/>
   <addaction name="action_Redo"/>
  </widget>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save &amp;As...</string>
   </property>
  </action>
  <action name="action_Export">
   <property name="text">
    <string>E&amp;xport...</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_New">
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionInsert_Edges">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/edit_edges.svg</normaloff>:/img/edit_edges.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Edges and Nodes</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+E</string>
   </property>
  </action>
  <action name="actionEdge_List">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/edge_list.svg</normaloff>:/img/edge_list.svg</iconset>
   </property>
   <property name="text">
    <string>Edge &amp;List</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+L</string>
   </property>
  </action>
  <action name="action_Link">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/link.svg</normaloff>:/img/link.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Link</string>
   </property>
   <property name="toolTip">
    <string>Link Selected Nodes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_Erase">
   <property name="text">
    <string>&amp;Erase</string>
   </property>
   <property name="toolTip">
    <string>Erase Selected Nodes</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_Merge">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/merge.svg</normaloff>:/img/merge.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Merge</string>
   </property>
   <property name="toolTip">
    <string>Merge Selected Nodes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="text">
    <string>Select &amp;All</string>
   </property>
  </action>
  <action name="actionRefresh_Path">
   <property name="text">
    <string>Refresh &amp;Path</string>
   </property>
  </action>
  <action name="action_Main">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Main</string>
   </property>
  </action>
  <action name="action_Knot_Editing">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Knot Editing</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom &amp;In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom &amp;Out</string>
   </property>
  </action>
  <action name="action_Reset_Zoom">
   <property name="text">
    <string>&amp;Reset Zoom</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="action_Reset_View">
   <property name="text">
    <string>&amp;Reset View</string>
   </property>
  </action>
  <action name="actionShow_Graph">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;Graph</string>
   </property>
  </action>
  <action name="actionShow_Knot_line">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Knot&amp;line</string>
   </property>
  </action>
  <action name="action_Style">
   <property name="text">
    <string>&amp;Style...</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
  </action>
  <action name="action_Copy">
   <property name="text">
    <string>&amp;Copy</string>
   </property>
  </action>
  <action name="action_Paste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
  </action>
  <action name="action_Configure_Knotter">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Pre&amp;ferences...</string>
   </property>
  </action>
  <action name="actionEnable_Grid">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/square_grid.svg</normaloff>:/img/square_grid.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Enable Grid</string>
   </property>
  </action>
  <action name="actionConfigure_G_rid">
   <property name="text">
    <string>Configure G&amp;rid...</string>
   </property>
  </action>
  <action name="action_Horizontal_Flip">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/hflip.svg</normaloff>:/img/hflip.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Horizontal Flip</string>
   </property>
  </action>
  <action name="action_Vertical_Flip">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/vflip.svg</normaloff>:/img/vflip.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Vertical Flip</string>
   </property>
  </action>
  <action name="action_Move_Grid">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/img/move_grid.svg</normaloff>:/img/move_grid.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Move Grid</string>
   </property>
  </action>
  <action name="actionForce_Crash">
   <property name="text">
    <string>Force &amp;Crash</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>KnotView</class>
   <extends>QGraphicsView</extends>
   <header>graphics/knotview.hpp</header>
   <slots>
    <signal>mouse_moved(QPointF)</signal>
    <slot>erase_selected()</slot>
    <slot>link_selected()</slot>
    <slot>merge_selected()</slot>
    <slot>select_all()</slot>
    <slot>redraw()</slot>
    <slot>reset_view()</slot>
    <slot>reset_zoom()</slot>
    <slot>flip_horizontal()</slot>
    <slot>flip_vertical()</slot>
    <slot>mode_move_grid()</slot>
    <slot>toggle_knotline(bool)</slot>
    <slot>toggle_graph(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Erase</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>erase_selected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Merge</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>merge_selected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Link</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>link_selected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSelect_All</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>select_all()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdge_List</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>mode_edge_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInsert_Edges</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>mode_edge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canvas</sender>
   <signal>mouse_moved(QPointF)</signal>
   <receiver>Knot_Window</receiver>
   <slot>mouse_moved(QPointF)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>495</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRefresh_Path</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>redraw()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_New</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Save</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Open</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Export</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>exportSVG()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Reset_View</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>reset_view()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Main</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Knot_Editing</sender>
   <signal>triggered(bool)</signal>
   <receiver>EditBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Reset_Zoom</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>reset_zoom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_In</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>zoom_in()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_Out</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>zoom_out()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCut</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Copy</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Paste</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConfigure_G_rid</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>configure_grid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEnable_Grid</sender>
   <signal>triggered(bool)</signal>
   <receiver>Knot_Window</receiver>
   <slot>enable_grid(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Horizontal_Flip</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>flip_horizontal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Vertical_Flip</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>flip_vertical()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Move_Grid</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>mode_move_grid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Graph</sender>
   <signal>triggered(bool)</signal>
   <receiver>canvas</receiver>
   <slot>toggle_graph(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Knot_line</sender>
   <signal>triggered(bool)</signal>
   <receiver>canvas</receiver>
   <slot>toggle_knotline(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionForce_Crash</sender>
   <signal>triggered()</signal>
   <receiver>Knot_Window</receiver>
   <slot>cause_crash()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>mode_edge_list()</slot>
  <slot>mode_edge()</slot>
  <slot>mode_node()</slot>
  <slot>mouse_moved(QPointF)</slot>
  <slot>clear()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>open()</slot>
  <slot>exportSVG()</slot>
  <slot>zoom_in()</slot>
  <slot>zoom_out()</slot>
  <slot>copy()</slot>
  <slot>cut()</slot>
  <slot>paste()</slot>
  <slot>enable_grid(bool)</slot>
  <slot>configure_grid()</slot>
  <slot>cause_crash()</slot>
 </slots>
</ui>
